File: src/qq_msgs/src/chao_node.cpp
#include<ros/ros.h>
#include<qq_msgs/Carry.h>
int main(int argc, char *argv[])
{
    ros::init(argc,argv,"chao_node");
    ros::NodeHandle nh;
    ros::Publisher chao_pub = nh.advertise<qq_msgs::Carry>("/kaiheiqun",1);
    qq_msgs::Carry msg;
    msg.grade = "王者";
    msg.star = 50;
    msg.data = "我带飞";
    ros::Rate loop_rate(10);

    while (ros::ok())
    {
        chao_pub.publish(msg);
        loop_rate.sleep();
    }
    return 0;
}

File: src/ssr_pkg/scripts/chao_node.py
#!/usr/bin/env python3
#coding=utf-8

import rospy
from std_msgs.msg import String
from qq_msgs.msg import Carry
def main():
    rospy.init_node("chao_node_py")
    pub = rospy.Publisher("kaiheiqunpy",Carry,queue_size=1)
    rate = rospy.Rate(10)
    while not rospy.is_shutdown():
        msg = Carry()
        msg.grade = "王者"
        msg.star = 50
        msg.data = "我带飞py"
        pub.publish(msg)
        rate.sleep()
if __name__ == '__main__':
    main()
    